dashboard/src/views/Projects/Versions/index.tsx

```diff
@@ -1,5 +1,143 @@
+import SearchIcon from "@mui/icons-material/Search";
+import {
+  Box,
+  Button,
+  FormControl,
+  FormLabel,
+  Input,
+  Sheet,
+  Table,
+  Typography,
+} from "@mui/joy";
+
+import { useEffect, useState } from "react";
+import { mockedVersionsResponse } from "../../../api/mocks/versions.mock";
+import { PaginationSection } from "../../../components/TablePaginationSection";
+import { TableRow } from "../../../components/TableRow";
+import { GetVersionsResponse } from "../../../interfaces/versions";
+
 export const VersionsOverview = () => {
-  return <div>VersionsOverview</div>;
-};
+  const [searchBar, setSearchBar] = useState("");
+  //TODO: replace mocked data with the response from the API when react query hooks are implemented
+  const versions = mockedVersionsResponse.versions;
+  const [versionsList, setVersionsList] = useState<GetVersionsResponse>(
+    mockedVersionsResponse,
+  );
+
+  useEffect(() => {
+    if (searchBar === "") {
+      return;
+    }
+    setVersionsList(versionsList);
+  }, [versionsList, searchBar]);
```

Comment:
this useEffect could cause an infinite loop. 

setVersionsList is updating the versionsList state in a way that triggers the effect again.

You could do something like this to avoid infinite loop

useEffect(() => {
  if (searchBar === "") {
    return;
  }

  setVersionsList((prevList) => {
    if (prevList !== versionsList) {
      return versionsList;
    }
    return prevList;
  });
}, [versionsList, searchBar]);

Reply:
copied that one